scripts/build_airocean_parameters.py

```diff
@@ -0,0 +1,560 @@
+        [-0.09755446, 0.22876301, -0.77481398],
+    ]
+)
+# Centers corresponding to the split faces around Japan
+ico_3_centers = np.array(
+    [
+        [-0.64642729, 0.48840818, -0.12653887],
+        [-0.45298488, 0.67661305, 0.09706879],
+        [-0.66606089, 0.42586898, 0.21776448],
+    ]
+)
+
+"""
+    Definition of the forward and inverse transform matrices from icosahedral faces to airocean unfolded net space
+"""
+
+A = ico_faces[:, 1] - ico_faces[:, 0]
+B = ico_faces[:, 2] - ico_faces[:, 0]
+ico_normals = -np.array(
+    [
+        A[:, 1] * B[:, 2] - A[:, 2] * B[:, 1],
+        A[:, 2] * B[:, 0] - A[:, 0] * B[:, 2],
+        A[:, 0] * B[:, 1] - A[:, 1] * B[:, 0],
+    ]
+).T
+ico_normals /= ((ico_normals**2).sum(axis=1) ** 0.5).reshape(-1, 1)
+ico_centers = ico_faces.mean(axis=1)
+ico_elevated_centers = ico_centers.copy()
+ico_elevated_centers[:, :3] += ico_normals
+
+airocean_basis = np.zeros((23, 4, 4))
+airocean_basis[:, :3, :] = airocean_23_face_vertices[:, :3, :]
+airocean_basis[:, 3, :] = airocean_23_centers
+
+ico_basis = np.zeros((23, 4, 4))
+ico_basis[:, :3, :] = ico_faces
+ico_basis[:, 3, :] = ico_elevated_centers
+
+
+air_ico_trans = np.array(
+    [
+        a.T @ np.linalg.inv(b.T)
+        for i, (a, b) in enumerate(zip(ico_basis, airocean_basis))
+    ]
+)
+ico_air_trans = np.array(
+    [
+        b.T @ np.linalg.inv(a.T)
+        for i, (a, b) in enumerate(zip(ico_basis, airocean_basis))
+    ]
+)
+
+for i, (a, b, v1, v2) in enumerate(
+    zip(ico_basis, airocean_basis, ico_centers, airocean_22_centers)
+):
+    m = b.T @ np.linalg.inv(a.T)  # ico to air
+    w = a.T @ np.linalg.inv(b.T)  # air to ico
+    for x, y in zip(a, b):
+        v1 = m @ x.T
+        v1 /= v1[3]
+        v2 = w @ y.T
+        v2 /= v2[3]
+        assert np.isclose(v1, y, rtol=1e-7).all()
+        assert np.isclose(v2, x, rtol=1e-7).all()
+
+    # Assert that the transform matrices work adequately
+    for i in range(100):
+        r = np.random.rand(3)
+        r /= r.sum()
+        v = (a[:3, :] * r.reshape(3, 1)).sum(axis=0)
+        v1 = m @ v.T
+        v1 /= v1[3]
+        v2 = w @ v1.T
+        v2 /= v2[3]
+        assert np.isclose(v, v2, rtol=1e-7).all()
+
+"""
+    End of definitions.
+"""
+
+
+def generate_20_faces_airocean_net(
+    outfile: Path = Path("./20_faces_airocean_net.png"),
+) -> None:
+    """
+    Save a picture of the "base" (non split) 20 faces of the unfolded icosahedron net
+    """
+    try:
+        import geopandas as gpd
+        from shapely.geometry import Polygon
+
+        gdf = gpd.GeoDataFrame(
+            data={"id": range(22)},
+            geometry=[Polygon(shell=v[:, :2]) for v in airocean_face_vertices],
+            crs=None,
+        )
+        ax = gdf.plot()
+        gdf.apply(
+            lambda x: ax.annotate(
+                text=x["id"], xy=x.geometry.centroid.coords[0], ha="center"
+            ),
+            axis=1,
+        )
+        ax.get_figure().savefig(str(outfile))
```

Comment:
`savefig` accepts `Path`s
```suggestion
        ax.get_figure().savefig(outfile)
```

Reply:
I thought it might depend on matplotlib version but you are correct